nilearn/datasets/tests/test_utils.py

```diff
@@ -275,37 +274,46 @@ def test_filter_columns():
 
 
 def test_uncompress():
-    # Create dummy file
-    fd, temp = mkstemp()
-    os.close(fd)
-
     # Create a zipfile
     dtemp = mkdtemp()
     ztemp = os.path.join(dtemp, 'test.zip')
     ftemp = 'test'
-    with contextlib.closing(zipfile.ZipFile(ztemp, 'w')) as testzip:
-        testzip.writestr(ftemp, 'test')
-    datasets.utils._uncompress_file(ztemp, verbose=0)
-    assert(os.path.exists(os.path.join(dtemp, temp)))
-    shutil.rmtree(dtemp)
-
-    dtemp = mkdtemp()
-    ztemp = os.path.join(dtemp, 'test.tar')
-    with contextlib.closing(tarfile.open(ztemp, 'w')) as tar:
-        tar.add(temp)
-    datasets.utils._uncompress_file(ztemp, verbose=0)
-    assert(os.path.exists(os.path.join(dtemp, temp)))
-    shutil.rmtree(dtemp)
-
-    dtemp = mkdtemp()
-    ztemp = os.path.join(dtemp, 'test.gz')
-    f = gzip.open(ztemp, 'wb')
-    f.close()
-    datasets.utils._uncompress_file(ztemp, verbose=0)
-    assert(os.path.exists(os.path.join(dtemp, temp)))
-    shutil.rmtree(dtemp)
-
-    os.remove(temp)
+    try:
+        with contextlib.closing(zipfile.ZipFile(ztemp, 'w')) as testzip:
+            testzip.writestr(ftemp, 'test')
+        datasets.utils._uncompress_file(ztemp, verbose=0)
+        assert (os.path.exists(os.path.join(dtemp, ftemp)))
+        shutil.rmtree(dtemp)
+
+        dtemp = mkdtemp()
+        ztemp = os.path.join(dtemp, 'test.tar')
+        ftemp = 'test'
+
+        # Create dummy file in the dtemp folder, so that the finally statement
```

Comment:
I could not understand the use of finally statement ? Anyway you are removing using ```shutil``` at each file extensions.